backend/src/baserow/core/two_factor_auth/registries.py

```diff
@@ -212,14 +219,27 @@ def verify(self, **kwargs) -> bool:
             raise VerificationFailed
 
         totp = pyotp.TOTP(provider.specific.secret)
+        current_ts = datetime.now(tz=timezone.utc)
+        hashed_code = hashlib.sha256(code.encode("utf-8")).hexdigest()
+
+        code_already_used = TOTPUsedCode.objects.filter(
+            user=provider.user, code=hashed_code
```

Comment:
What do you think of adding a `db_index=True` to the `code` property? It doesn't look like it's going to be a problem, but if there are many users with a TOTP code in the instance, it could slow things down a little bit.